float32[] data
int32 height
int32 width
